### PR TITLE
PUB-927 - Court Reference Data

### PR DESCRIPTION
**Pull request by ChrisS1512**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/PUB-927

### Change description ###

Added CSV of reference data, alongside updating of the endpoints to retrieve it.

Key things to highlight:

1) The Court model has changed, to handle multiple jurisidctions, provenance references and also the removal of hearings being part of the model
2) The /courts/find/courtName> endpoint has changed to /courts/name/<courtName>
3) The filter endpoint now takes params rather than a body. It accepts two params, which are comma seperated arrays - "jurisidctions" and "region"
4) Addition of the POST endpoint which takes an uploaded CSV. See resources folder for examples.

NOTE - This PR will break the frontend. Will need to merge this through at the same time as ticket 1030

**Does this PR introduce a breaking change?** (check one with "x")

```
[x] Yes
[ ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/pip/data/management/models/court/CourtCsv.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+ * This class captures the inbound Court CSV that will be ingested.
+ */
+@Data
+public class CourtCsv {
+
+    @CsvBindByName(column = "P&I ID")
+    private Integer uniqueId;
+
+    @CsvBindByName(column = "Court Desc")
+    private String courtName;
```
Comment (junaidiqbalmoj) [inline src/main/java/uk/gov/hmcts/reform/pip/data/management/models/court/CourtCsv.java:20]: Can you please change **courtName** to **name**? Frontend is using **name**.

Comment (ChrisS1512) [reply src/main/java/uk/gov/hmcts/reform/pip/data/management/models/court/CourtCsv.java:20]: Updated to name :)

File: src/main/java/uk/gov/hmcts/reform/pip/data/management/models/court/Court.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    name = "list-array",
+    typeClass = ListArrayType.class
+)
+public class Court {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    @Column(columnDefinition = "uuid", insertable = false, updatable = false, nullable = false)
+    @Type(type = "org.hibernate.type.PostgresUUIDType")
+    private UUID courtId;
```
Comment (junaidiqbalmoj) [inline src/main/java/uk/gov/hmcts/reform/pip/data/management/models/court/Court.java:41]: Can you please check it to Integer? Frontend is using courtId as INT in models and other places as well.

**Review by junaidiqbalmoj — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/reform/pip/data/management/models/court/Court.java (hunk 2 above)
Comment (ChrisS1512) [reply src/main/java/uk/gov/hmcts/reform/pip/data/management/models/court/Court.java]: Updated :) Had a discussion, and UUID was a bit excessive especially when the URL's are going to display the UUID.

**Review by furnivall — APPROVED** [review]
👍

**Review by furnivall — DISMISSED** [review]
👍

**Review by junaidiqbalmoj — DISMISSED** [review]
as discussed, everything looks good 👍

**Review by alberto-hortiguela — APPROVED** [review]

**Review by furnivall — APPROVED** [review]
LGTM

**Pull request merged**